schematic/store/synapse.py

```diff
@@ -659,105 +670,36 @@ def getFilesInStorageDataset(
-            current_location = self.synapse_entity_tracker.get(
-                synapse_id=current_location["parentId"],
-                syn=self.syn,
-                download_file=False,
-            )
-            return walk_back_to_project(
-                current_location=current_location,
-                location_prefix=updated_prefix,
-                skip_entry=False,
-            )
+        # The query will also be ammended to include everything containted in all the subdirectories of the dataset
+        for subfolder, _, path in folders:
+            dataset_clause += f"OR parentId='{subfolder[1]}' "
+        dataset_clause = f"({dataset_clause})"
 
-        prefix = walk_back_to_project(
-            current_location=current_entity_location,
-            location_prefix="",
-            skip_entry=True,
-        )
+        # When querying, only include files to exclude entity files and subdirectories
+        where_clauses = [dataset_clause, "type='file'"]
 
-        project_id = self.getDatasetProject(datasetId)
-        project = self.synapse_entity_tracker.get(
-            synapse_id=project_id, syn=self.syn, download_file=False
-        )
-        project_name = project.name
-        file_list = []
+        # Requery the fileview to specifically get the files in the given dataset
+        self.query_fileview(columns=["id", "path"], where_clauses=where_clauses)
 
-        # iterate over all results
-        for dirpath, _, path_filenames in walked_path:
-            # iterate over all files in a folder
-            for path_filename in path_filenames:
-                if ("manifest" not in path_filename[0] and not fileNames) or (
-                    fileNames and path_filename[0] in fileNames
-                ):
-                    # don't add manifest to list of files unless it is specified in the
-                    # list of specified fileNames; return all found files
-                    # except the manifest if no fileNames have been specified
-                    # TODO: refactor for clarity/maintainability
-
-                    if fullpath:
-                        # append directory path to filename
-                        if dirpath[0].startswith(f"{project_name}/"):
-                            path_without_project_prefix = (
-                                dirpath[0] + "/"
-                            ).removeprefix(f"{project_name}/")
-                            path_filename = (
-                                prefix + path_without_project_prefix + path_filename[0],
-                                path_filename[1],
-                            )
-                        else:
-                            path_filename = (
-                                prefix + dirpath[0] + "/" + path_filename[0],
-                                path_filename[1],
-                            )
+        # Exclude manifest files
+        non_manifest_files = self.storageFileviewTable.loc[
```

Comment:
@thomasyu888 I think we could note that higher up in the docstring instead of under one of the parameters and in the description of the `storage/dataset/files` endpoint.
as an aside, do we know of any users that don't use the default basename and change it in their configs? I guess we could also just check for both

Reply:
@GiaJordan good point, for now, I searched "basename" in the data_curator_config: https://github.com/search?q=repo%3ASage-Bionetworks%2Fdata_curator_config%20basename&type=code and didn't get anything returned, but we'll probably want to watch for that eventually to pull from the basename config instead of using "manifest" because people can technically change that.

Can you help create a ticket for that to track?

Reply:
Yes I can. It could also be technically changed locally in the `config.yml` but I'm not aware of projects that change that parameter currently